File: works/1011.cpp
#include <iostream>
#include <cmath>
using namespace std;

// 남은 거리, 바로전에 이동한 거리, 카운트 
long long dfs(long long distance, long long moved, long long cnt)
{
    if(distance == 0) return cnt;
//    cout << "dis: " << distance << ' ' << "moved: " << moved << ' ' << "cnt: " << cnt << endl;

    for(long long move = moved+1; move >= moved-1; move--)
    {
        if(move <= 0) continue;
        long long sum = (move * (move + 1)) / 2;
        if(sum <= distance)
        {
            return dfs(distance-move, move, cnt+1);
        }
    }
}

int main()
{
    ios::sync_with_stdio(false); cin.tie(NULL);
    int t; cin >> t;
    while(t--)
    {
        long long x, y; cin >> x >> y;
        long long distance = y - x;
        long long ans = dfs(distance, 0, 0);
        cout << ans << '\n';
    }

}

File: works/1011.kt
import java.io.BufferedReader
import java.io.InputStreamReader
import java.util.*

fun dfs(distance:Long, moved:Long, cnt:Long): Long
{
    if(distance == 0L) return cnt

    for(move in moved+1 downTo moved-1)
    {
        if(move <= 0) continue
        var sum = (move * (move + 1)) / 2
        if(sum <= distance) return dfs(distance-move, move, cnt+1)
    }
    return 0
}

fun main()
{
    val br = BufferedReader(InputStreamReader(System.`in`))
    var t = br.readLine().toInt()

    while(t-- > 0)
    {
        val _s = br.readLine().split(' ')
        var x = _s[0].toInt()
        var y = _s[1].toInt()
        var distance = (y - x).toLong()
        var ans = dfs(distance, 0, 0)
        println(ans)
    }
}